backend/src/bundles/users/user.service.ts

```diff
@@ -27,20 +39,37 @@ class UserService implements IService {
         };
     }
 
-    public async create(
-        payload: UserSignUpRequestDto,
-    ): Promise<UserSignUpResponseDto> {
+    public async create({
+        email,
+        password,
+        firstName,
+        lastName,
+    }: UserSignUpRequestDto): Promise<Pick<UserEntityFields, 'id'>> {
+        const { hash: passwordHash, salt: passwordSalt } =
+            encryptPassword(password);
```

Comment:
It would be better if the user service only accepted user data for further saving. And the logic associated with generating tokens, checking passwords, etc. was at the auth service